frontend/src/stimulus/controllers/async-dialog.controller.ts

```diff
@@ -49,13 +49,20 @@ export default class AsyncDialogController extends ApplicationController {
         Accept: 'text/vnd.turbo-stream.html',
         'X-Authentication-Scheme': 'Session',
       },
-    }).then((r) => r.text())
-      .then((html) => {
-        renderStreamMessage(html);
-      })
-      .finally(() => {
-        TurboHelpers.hideProgressBar();
-      });
+    }).then((response) => {
+      const contentType = response.headers.get('Content-Type') || '';
+      const isTurboStream = contentType.includes('text/vnd.turbo-stream.html');
```

Comment:
rThe typings for `@hotwired/turbo` are currently broken (I started making fixes here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71699 but didn't finish)

This should be possible though:

```suggestion
      const isTurboStream = contentType === StreamMessage.contentType
```

Reply:
I guess I can use `StreamMessage.contentType` in the `includes` call, but not as you suggested, because `contentType` is the full HTTP header, which includes the charset as well, e.g.

```
console.log(contentType);
# => text/vnd.turbo-stream.html; charset=utf-8
```

On the other hand, `StreamMessage.contentType` [is the content type only](https://github.com/hotwired/turbo/blob/main/src/core/streams/stream_message.js#L4): `text/vnd.turbo-stream.html`

I guess we could (in a separate commit) replace all occurences of the content type in our code (19 results in 17 files) by the constant. I just have to figure out how JS imports work :sweat_smile:

Reply:
@myabc So my JS is not really good, but reading https://github.com/hotwired/turbo/blob/main/src/index.js and considering that `StreamMessage` is defined in `core/streams`, is it possible that I can't import it from `@hotwired/turbo` at all? All things that I can successfully import are listed in the index file (or at least globbed).

Reply:
@NobodysNightmare I dug into the source, and you're quite right - it's not exported. Had the Turbo npm maintainers included the sources in addition to the ESM bundle, then it might have been possible to `import "@turbo/core/streams/stream-message"` directly - but this is brittle since you're basically relying on a private API.

Reply:
> I guess I can use StreamMessage.contentType in the includes call, but not as you suggested, because contentType is the full HTTP header, which includes the charset as well

I think the MIME type always precedes the charset.

```suggestion
      const isTurboStream = contentType.startsWith('text/vnd.turbo-stream.html');
```